add documentation to lektorium config file (#55)
sphericalpm/lektorium: DEPLOYMENT.md

# Quickstart guide

> Current implementation of hosted authoring central supports different method of storing configuration and sites sources, execution environments. Here we will discuss only main one - everything will be stored in GitLab repositories and executed in docker containers.

Reviewer: ```suggestion
The current implementation of the hosted authoring central supports different methods of storing configuration of site sources and execution environments. Here we will discuss only main one: everything stored in GitLab repositories and executed in docker containers.
```

> Lektorium server needs separate repository to store its configuration. It will fetch this repository on start and use `config.yml` from repository root as registry of managed sites. The format of this registry file is described in separate section at the end of this article.

Reviewer: ```suggestion
Lektorium server needs a separate repository to store its configuration. It fetches this repository on start and uses `config.yml` from repository root as a registry of managed sites. The format of this registry file is described in a separate section at the end of this document.
```

> We will use docker to start lektorium, reverse proxy server and lektor instances. For GitLab access lektorium uses ssh public key authentication so you need to generate keypair, put private part to `containers/server/key` and configure GitLab to allow access to lektorium and websites repositories with this key.

Reviewer: ```suggestion
Docker is used to start lektorium, reverse proxy server and lektor instances. For GitLab access, lektorium uses ssh public key authentication so one needs to generate a keypair, place private key to `containers/server/key` and configure GitLab to allow access to lektorium and websites' repositories with this key.
```

## Install and start

> You need to create config for you server in `invoke.yaml` file.

Reviewer: ```suggestion
You need to create a deployment options file for you server in the `invoke.yaml` file.
```

  GIT_USER: Lektorium
```

> * `server-name` option used by reverse proxy container;

Reviewer: ```suggestion
* `server-name` host name of the hosted authoring central
```

> * `auth` option configures auth0 provider and described below in separate section;

Reviewer: ```suggestion
* `auth`  auth0 provider configuration (described in a separate section)
```

> * `cfg` option can be adopted with proper configuration repository path;

Reviewer: ```suggestion
* `cfg` may be used to locate the configuration repository
```

> * `network` option configure lektorium with docker network to be used to start containers, pay attention that it can not be default docker network because name resolution is not working with it;

Reviewer: ```suggestion
* `network` docker network to be used to start containers. Pay attention that it **can not** be the default docker network.
```

> * `env` is a list of additional environment options, where GIT_MAIL and GIT_USER options are mandatory for the proper work of git client.

Reviewer: ```suggestion
* `env` additional environment options. `GIT_MAIL` and `GIT_USER` options are mandatory for the proper work of the git client.
```

Deploy and start server:
```
inv -pe build run
```

Deploy and start reverse proxy:
```
inv -pe build-proxy-image run-proxy
```

## Auth0 as authentication provider
`auth` option of `invoke.yaml` configuration file consists of comma separated options of your auth0 application: domain, client id and api name. Please check auth0 documentation for more information or start lektorium **without authentication** by adding empty auth option to command line:
```
inv -pe build run --auth=''
```

> ## Sites configuration file

Reviewer: ```suggestion
## Site configuration file
```
